lib/ansible/module_utils/basic.py

```diff
@@ -1421,6 +1434,15 @@ def _set_defaults(self, pre=True):
                 if k not in self.params:
                     self.params[k] = default
 
+    def _set_fallbacks(self):
+        for k,v in self.argument_spec.items():
+            fallback_strategy, fallback_parameters = v.get('fallback', (None,[]))
```

Comment:
We probably want a third parameter for keyword args.  Some fallback methods might want the ability to use optional parameters.